sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -67,83 +56,57 @@ public static SentryClient init(SentryClientFactory sentryClientFactory) {
      *
      * @param dsn Data Source Name of the Sentry server.
      * @return SentryClient
-     */
-    public static SentryClient init(String dsn) {
-        return init(dsn, null);
-    }
-
-    /**
-     * Initialize and statically store a {@link SentryClient} by using the provided
-     * {@link Dsn} and {@link SentryClientFactory}.
-     * <p>
-     * Note that the Dsn or SentryClientFactory may be null, at which a best effort attempt
-     * is made to look up or choose the best value(s).
      *
-     * @param dsn                 Data Source Name of the Sentry server.
-     * @param sentryClientFactory SentryClientFactory to use.
-     * @return SentryClient
+     * @deprecated use {@link #init(SentryOptions)}
      */
-    public static SentryClient init(String dsn, SentryClientFactory sentryClientFactory) {
-        SentryOptions sentryOptions = new SentryOptions();
-        sentryOptions.setDsn(dsn);
-        sentryOptions.setSentryClientFactory(sentryClientFactory);
-        return init(sentryOptions);
+    @Deprecated
+    public static SentryClient init(@Nullable String dsn) {
```

Comment:
Is it also needed to deprecate the main init function to get rid of testNG? Maybe this could be a separate PR so we can discuss?